### PR TITLE
Parse properly error message when changing cluster version with cluster.sh

### LINKED ISSUE #11620: Wrong message provided by cluster.sh for failed cluster version upgrade
When I try a rolling upgrade `3.8.6->3.9`, with the `cluster.sh` script and some member is still running 3.8.6, then the scripts output is:

```bash
$ ./cluster.sh -o change-cluster-version -v 3.9 -a 172.17.0.4
No port is defined, running script with default port : '5701'.
No groupname is defined, running script with default groupname : 'dev'.
No password is defined, running script with default password : 'dev-pass'.
Changing cluster version to 3.9 on ip 172.17.0.4 on port 5701
Cluster is in {, change failed.
```

The last printed line is wrong because an incorrect `sed` expresions are used to parse a version from the server response. IMO the message received from server should be used instead. The server response seems like:

```json
{"status":"fail","message":"Node's codebase version 3.8.6 is incompatible with the requested cluster version 3.9"}
```

### PR DESCRIPTION
**Pull request by vbekiaris**
Fixes #11620

### CODE REVIEW

File: hazelcast/src/main/resources/cluster.sh [hunk 1]
```diff
@@ -262,9 +262,10 @@ if [ "$OPERATION" = "change-cluster-version" ]; then
...
     if [ "$STATUS" = "fail" ];then
-       NEWVERSION=$(echo "${response}" | sed -e 's/^.*"version"[ ]*:[ ]*"//' -e 's/".*//');
-       if [ "$NEWVERSION" != "null" ]; then
-           echo "Cluster is in ${NEWVERSION}, change failed."
+       MESSAGE=$(echo "${response}" | sed -e 's/^.*"message"[ ]*:[ ]*"//' -e 's/".*//');
+       # when there is no "message" element in response, above sed process returns "{"
+       if [ "$MESSAGE" != "{" ]; then
+           echo "Cluster version change failed: ${MESSAGE}"
        else
             echo "An error occured while changing cluster version!";
```
Comment (Donnerbart) [inline hazelcast/src/main/resources/cluster.sh]: Shouldn't we show the response here? It seems quite informative, if I understand the issue correctly.

Comment (vbekiaris) [reply hazelcast/src/main/resources/cluster.sh]: If there was a `"message"` element in the returned JSON (which [should be always the case](https://github.com/vbekiaris/hazelcast/blob/673f032883917e6905e496fb80fcbaba5e713487/hazelcast/src/main/java/com/hazelcast/internal/ascii/rest/HttpPostCommandProcessor.java#L518-L520)), then the message value will be displayed. This branch is handling non-standard response (which may or may not be informative ;-) ). Should we show the complete response in this case?

**Comment by vbekiaris** [comment]
run-lab-run

File: hazelcast/src/main/resources/cluster.sh (hunk 1 above)
Comment (Donnerbart) [reply hazelcast/src/main/resources/cluster.sh]: I understood the issue that the regex is wrong, so the version cannot be parsed. Not that there is no message at all.

**Pull request merged**